Feedback analysis scatterplots

forsyth2 opened the pull request:
Second component of ENSO Diagnostics: feedback analysis scatterplots

forsyth2 commented on acme_diags/driver/enso_diags_driver.py:
```diff
@@ -233,30 +243,99 @@ def run_diag(parameter):
+    # x['region'] = parameter.nino_region
+    # y['region'] = parameter.regions[0]
+    x = {'var': 'TS', 'units': 'degC', 'region': 'NINO3'}
+    # todo: select same way nino region, average over the region selected and that becomes a point in the monthly
```
Remove comment

forsyth2 commented on acme_diags/viewer/enso_diags_viewer.py:
```diff
@@ -20,47 +20,55 @@ def create_viewer(root_dir, parameters):
+                    relative_path + nc_ext for nc_ext in ['_test.nc', '_ref.nc', '_diff.nc']]
+                formatted_files = [
+                    {'url': f, 'title': f} for f in nc_files]
+            # TODO: will param.variables ever be longer than one variable?
```
Address this comment - perhaps in another issue/pull request?

forsyth2 replied:
Will address in a future pull request if necessary.

forsyth2 commented on run.py:
```diff
@@ -0,0 +1,23 @@
+import os
```
Probably will not include this file when merging.

forsyth2 commented on scatter.cfg:
```diff
@@ -0,0 +1,35 @@
+[#]
```
Remove this file before merging.

forsyth2 commented on tests/system/all_sets.cfg:
```diff
@@ -165,3 +165,21 @@ ref_file = 'TREFHT_201201_201312.nc'
+[#]
+sets = ["enso_diags"]
+plot_type = "scatter"
+case_id = "TREFHT"
```
Is `TREFHT` sufficient for tests?

chengzhuzhang replied:
Yes

forsyth2 commented:
@chengzhuzhang Ready for code review. I made some comments above. Also, I will look into possibly splitting the regression lines in the scatterplots. I copied plots generated on my laptop to https://portal.nersc.gov/project/acme/forsyth/feedback_analysis/.

chengzhuzhang commented on acme_diags/driver/enso_diags_driver.py:
```diff
@@ -233,30 +243,99 @@ def run_diag(parameter):
+                y['units'] = '10^3 {}'.format(y['units'])
+            parameter.var_id = '{}-feedback'.format(y['var'])
+            title_tuple = (y['var'], y['region'], x['var'], x['region'])
+            parameter.main_title = 'Feedback, {} anomaly ({}) vs. {} anomaly ({})'.format(*title_tuple)
```
Change `Feedback` to `Regression` to be main title and display in description in viewer

forsyth2 replied:
On line 194, we set the main title to be "Regression coefficient" for the maps. Wouldn't it be slightly confusing to also have the scatterplots titled "Regression"? Then again, I suppose they are both displaying regressions.

chengzhuzhang replied:
ah! I edited it moment ago. Removing 'Feedback, 'would be good.

forsyth2 replied:
Ah ok sorry, sounds good

forsyth2 replied:
@chengzhuzhang Is it ok to leave `-feedback` two lines above for `var_id`?

forsyth2 replied:
Leaving `feedback` in `var_id` and `output_file`

chengzhuzhang commented on scatter.cfg:
```diff
@@ -0,0 +1,35 @@
+[#]
+sets = ["enso_diags"]
+plot_type = "scatter"
+case_id = "ocean-zonal-wind-stress"
```
change `TAUX-feedback` so that they are consistent with the *.cfg files for regression coefficient one.

forsyth2 commented:
@chengzhuzhang Addressed code review comments and updated regression lines. Code generated on my Mac can be found at [`python run.py`](https://portal.nersc.gov/project/acme/forsyth/feedback_analysis/feedback_analysis/enso_reg_diags_model_to_model_local/viewer/enso_diags/index.html) and [`./tests/test.sh`](https://portal.nersc.gov/project/acme/forsyth/feedback_analysis/feedback_analysis/all_sets_results_test/viewer/enso_diags/index.html)

chengzhuzhang commented:
@forsyth2 The results looking good [here](https://portal.nersc.gov/project/acme/forsyth/feedback_analysis/feedback_analysis/enso_reg_diags_model_to_model_local/viewer/enso_diags/index.html)! I only have one comment, when linear regression was separated for positive and negative TS anomaly, the legend boxes block large portion of the plots. I would following the legends put in for single regression line to reduce the texts in the box, hopefully to reduce the area of the box. In addition, set y axis range greater may also help mitigate the overlapping problem?

forsyth2 commented:
Posted results of `run.py` to https://portal.nersc.gov/project/acme/forsyth/feedback_analysis_2020_03_18/ (viewer at https://portal.nersc.gov/project/acme/forsyth/feedback_analysis_2020_03_18/enso_reg_diags_model_to_model_local/viewer/enso_diags/index.html)
